CI Failures: 80_data_frame_jobs_crud

These tests have failed for quite a while on 7.x:

https://groups.google.com/a/elastic.co/forum/#!searchin/build-elasticsearch/80_data_frame_jobs_crud%7Csort:date

Don't think we can mute yml tests though?

`_data_frame/transforms/old-simple-continuous-transform/_stops` times out after the previous `_start` call fails due to a missing index.

This shows TWO bugs, the predicate for _stop was not satisfied even though the task was indeed cancelled and removed, we are failing to create a checkpoint in a mixed cluster.

The first bug has probably a couple of causes:
* An exception occurred when trying to stop it and since we ignore the listener here:
Exists in 7.3 and 7.x and master:
https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportStopTransformAction.java#L198..L201

we just never see the next cluster state update as it doesn't occur.

* A cluster update just does not fire and the predicate never gets called...This has happened in tests before as there are few cluster state updates occurring.

The error trace (when running in a mixed 7.3.0, 7.5.0 cluster):

```
[o.e.x.t.t.ClientTransformIndexer] [v7.3.0-0] [old-simple-continuous-transform] failed to create checkpoint.
»  org.elasticsearch.index.IndexNotFoundException: no such index [.transform-internal-003]
»       at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:194) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteWriteIndex(IndexNameExpressionResolver.java:304) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$ConcreteIndices.resolveIfAbsent(TransportBulkAction.java:652) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$BulkOperation.addFailureIfIndexIsUnavailable(TransportBulkAction.java:606) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$BulkOperation.doRun(TransportBulkAction.java:449) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction.executeBulk(TransportBulkAction.java:632) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$1.lambda$onResponse$0(TransportBulkAction.java:246) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:703) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
»       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
»       at java.lang.Thread.run(Thread.java:834) [?:?]
» WARN ][o.e.x.t.t.ClientTransformIndexer] [v7.3.0-0] [old-simple-continuous-transform] transform encountered an exception:
»  java.lang.RuntimeException: Failed to create checkpoint due to no such index [.transform-internal-003]
»       at org.elasticsearch.xpack.transform.transforms.ClientTransformIndexer.lambda$createCheckpoint$15(ClientTransformIndexer.java:502) [transform-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.ActionListener$1.onFailure(ActionListener.java:71) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.support.TransportAction$1.onFailure(TransportAction.java:79) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.support.ContextPreservingActionListener.onFailure(ContextPreservingActionListener.java:50) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportSingleItemBulkWriteAction.lambda$wrapBulkResponse$0(TransportSingleItemBulkWriteAction.java:64) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$BulkOperation.doRun(TransportBulkAction.java:502) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction.executeBulk(TransportBulkAction.java:632) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$1.lambda$onResponse$0(TransportBulkAction.java:246) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:703) [elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
»       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
»       at java.lang.Thread.run(Thread.java:834) [?:?]
»  Caused by: org.elasticsearch.index.IndexNotFoundException: no such index [.transform-internal-003]
»       at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteIndices(IndexNameExpressionResolver.java:194) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.cluster.metadata.IndexNameExpressionResolver.concreteWriteIndex(IndexNameExpressionResolver.java:304) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$ConcreteIndices.resolveIfAbsent(TransportBulkAction.java:652) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$BulkOperation.addFailureIfIndexIsUnavailable(TransportBulkAction.java:606) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       at org.elasticsearch.action.bulk.TransportBulkAction$BulkOperation.doRun(TransportBulkAction.java:449) ~[elasticsearch-7.5.0-SNAPSHOT.jar:7.5.0-SNAPSHOT]
»       ... 7 more
```

This occurred in the `oneThirdUpgrade` test.

Don't let the trace fool you, this `7.3.0-0` was upgraded to `7.5.0` and joined the cluster. The current master node was a 7.3.0 node.

The transform task was then assigned to the new 7.5.0 node and it attempted to create a new checkpoint since it is attempting to `_start` with no previous checkpoint data.

--- a/x-pack/qa/rolling-upgrade/build.gradle
+++ b/x-pack/qa/rolling-upgrade/build.gradle
@@ -146,13 +146,12 @@ for (Version bwcVersion : bwcVersions.wireCompatible) {
                 'mixed_cluster/80_data_frame_jobs_crud/Test put batch data frame transforms on mixed cluster',
                 'mixed_cluster/80_data_frame_jobs_crud/Test put continuous data frame transform on mixed cluster'
         ]
-        // Dataframe transforms were not added until 7.2.0
-        if (bwcVersion.before('7.2.0')) {
-            toBlackList << 'mixed_cluster/80_data_frame_jobs_crud/Test GET, start, and stop old cluster batch transforms'
-        }
-        // continuous Dataframe transforms were not added until 7.3.0
-        if (bwcVersion.before('7.3.0')) {
-            toBlackList << 'mixed_cluster/80_data_frame_jobs_crud/Test GET, stop, start, old continuous transforms'
+        // transform in mixed cluster is effectively disabled till 7.4, see gh#48019
+        if (bwcVersion.before('7.4.0')) {
+            toBlackList.addAll([
+                'mixed_cluster/80_data_frame_jobs_crud/Test GET, start, and stop old cluster batch transforms',
+                'mixed_cluster/80_data_frame_jobs_crud/Test GET, stop, start, old continuous transforms'
+            ])
         }
         systemProperty 'tests.rest.blacklist', toBlackList.join(',')
     }
@@ -168,15 +167,14 @@ for (Version bwcVersion : bwcVersions.wireCompatible) {
         systemProperty 'tests.rest.suite', 'mixed_cluster'
         systemProperty 'tests.first_round', 'false'
         def toBlackList = []
-        if (bwcVersion.before('7.2.0')) {
+        // transform in mixed cluster is effectively disabled till 7.4, see gh#48019
+        if (bwcVersion.before('7.4.0')) {
             toBlackList.addAll([
                     'mixed_cluster/80_data_frame_jobs_crud/Test put batch data frame transforms on mixed cluster',
                     'mixed_cluster/80_data_frame_jobs_crud/Test GET, start, and stop old cluster batch transforms',
-                    'mixed_cluster/80_data_frame_jobs_crud/Test put continuous data frame transform on mixed cluster'
+                    'mixed_cluster/80_data_frame_jobs_crud/Test put continuous data frame transform on mixed cluster',
+                    'mixed_cluster/80_data_frame_jobs_crud/Test GET, stop, start, old continuous transforms'
             ])
-        }
-        if (bwcVersion.before('7.3.0')) {
-            toBlackList << 'mixed_cluster/80_data_frame_jobs_crud/Test GET, stop, start, old continuous transforms'
         }
         if (!toBlackList.empty) {
             systemProperty 'tests.rest.blacklist', toBlackList.join(',')

